### PR TITLE
agent: move vmEventAdded handling into its own method

### PR DESCRIPTION
**Pull request by sharnoff**

### CODE REVIEW

File: pkg/agent/globalstate.go [hunk 1]
```diff
@@ -94,64 +94,74 @@ func (s *agentState) handleEvent(ctx context.Context, event vmEvent) {
...
+		errored:  nil,
+		panicked: false,
+		vmInfo:   event.vmInfo,
+	}
+
+	runner := &Runner{
+		global: s,
+		status: status,
+		logger: RunnerLogger{
+			prefix: fmt.Sprintf("Runner %v: ", event.podName),
```
Comment (tychoish) [inline pkg/agent/globalstate.go:129]: what's the type of the podname and why is this a %v and not an %s

File: pkg/agent/globalstate.go [hunk 2]
```diff
@@ -94,64 +94,74 @@ func (s *agentState) handleEvent(ctx context.Context, event vmEvent) {
...
+		return
+	}
+
+	runnerCtx, cancelRunnerContext := context.WithCancel(ctx)
+
+	status := &podStatus{
+		lock:     sync.Mutex{},
+		done:     false,
+		errored:  nil,
+		panicked: false,
```
Comment (tychoish) [inline pkg/agent/globalstate.go:121]: I still think that declaring all these zero values is a bit unidiomatic, and makes things harder to read.

File: pkg/agent/globalstate.go [hunk 3]
```diff
@@ -94,64 +94,74 @@ func (s *agentState) handleEvent(ctx context.Context, event vmEvent) {
...
 }
 
+func (s *agentState) handleVMEventAdded(
+	ctx context.Context,
+	event vmEvent,
+	podName util.NamespacedName,
+) {
+	state, hasPod := s.pods[podName]
+
+	if hasPod {
```
Comment (tychoish) [inline pkg/agent/globalstate.go:110]: the `hasPod` vs `ok` for the name of the existence check threw me for a second.

**Review by tychoish — APPROVED** [review]

File: pkg/agent/globalstate.go (hunk 1 above)
Comment (sharnoff) [reply pkg/agent/globalstate.go:129]: Oh good catch. This was previously a `util.NamespacedName`, but got changed to `string` by a recent PR. Opened #159

**Pull request merged**